server/fleet/teams.go

```diff
@@ -277,28 +264,46 @@ func ValidateRole(globalRole *string, teamUsers []UserTeam) error {
 	return nil
 }
 
-func ValidateRoleForLicense(globalRole *string, teamUsers *[]UserTeam, license LicenseInfo) error {
+// ValidateUserRoles verifies the roles to be applied to a new or existing user.
+//
+// Argument createNew sets whether the user is being created (true) or is being modified (false).
+func ValidateUserRoles(createNew bool, payload UserPayload, license LicenseInfo) error {
 	var teamUsers_ []UserTeam
-	if teamUsers != nil {
-		teamUsers_ = *teamUsers
+	if payload.Teams != nil {
+		teamUsers_ = *payload.Teams
 	}
-	if err := ValidateRole(globalRole, teamUsers_); err != nil {
+	if err := ValidateRole(payload.GlobalRole, teamUsers_); err != nil {
 		return err
 	}
 	premiumRolesPresent := false
-	if globalRole != nil {
-		if _, ok := premiumGlobalRoles[*globalRole]; ok {
+	gitOpsRolePresent := false
+	if payload.GlobalRole != nil {
+		if *payload.GlobalRole == RoleGitOps {
+			gitOpsRolePresent = true
+		}
+		if _, ok := premiumGlobalRoles[*payload.GlobalRole]; ok {
 			premiumRolesPresent = true
 		}
 	}
 	for _, teamUser := range teamUsers_ {
+		if teamUser.Role == RoleGitOps {
+			gitOpsRolePresent = true
+		}
 		if _, ok := premiumTeamRoles[teamUser.Role]; ok {
 			premiumRolesPresent = true
 		}
 	}
 	if !license.IsPremium() && premiumRolesPresent {
 		return ErrMissingLicense
 	}
+	if gitOpsRolePresent &&
+		// New user is not API only.
+		((createNew && (payload.APIOnly == nil || !*payload.APIOnly)) ||
```

Comment:
I'm guessing that if `payload.APIOnly == nil` conceptually is the same as having `payload.APIOnly = false`? (i.e. omission is assumed to be the same as a false value. If so, maybe we can simplify this to be just:
```
gitOpsRolePresent && payload.APIOnly != nil && *payload.APIOnly
```

Reply:
> I'm guessing that if payload.APIOnly == nil conceptually is the same as having payload.APIOnly = false?

During user creation (POST), yes.
During user modification (PATCH), it means such field is omitted and should not be updated.